deepgrp/__main__.py

```diff
@@ -30,164 +30,283 @@ def read_multi_fasta(filestream: TextIO) -> Iterator[Tuple[str, str]]:
+        _LOG.setLevel(level=loglevel)
+        tf.debugging.set_log_device_placement(loglevel == logging.DEBUG)
+        return self
+
+    def run(self):
+        """Run the command provided by the user."""
+        options = dgmodel.Options(
+            min_mss_len=self.args.min_mss_length,
+            batch_size=self.args.batch_size,
+            xdrop_len=self.args.xdrop_length,
+        )
+        getattr(self, self.args.command)(self.args, options)
+
+    @staticmethod
+    def predict(args: argparse.Namespace, options: dgmodel.Options):
+        """Predict with deepgrp.
+
+        Args:
+            args (argparse.Namespace): Command line arguments.
+                Includes: 'model', 'output', 'FASTA', 'no_use_mss',
+                'step_size'
+            options (dgmodel.Options): Default and user provided options.
+        """
+
+        _LOG.debug("Loading model %s!", args.model)
+        model = tf.keras.models.load_model(
+            args.model,
+            custom_objects={
+                "ReverseComplement": dgmodel.ReverseComplement,
+            },
+        )
+        options.vecsize = model.input_shape[1]
+        _LOG.info("Model loading finished successfully!")
+        outstream = sys.stdout if args.output == "-" else open(args.output, "w")
+
+        for filename in args.FASTA:
+            _LOG.info("Processing %s", filename)
+            try:
+                filestream = sys.stdin if filename == "-" else open(filename, "r")
+                for header, dnasequence in _read_multi_fasta(filestream):
+                    predictions, startpos = _predict(
+                        dnasequence,
+                        model,
+                        options,
+                        args.step_size,
+                        use_mss=not args.no_use_mss,
+                    )
+                    for segment in dgsequence.yield_segments(predictions, startpos):
+                        if segment[2] > 0:
+                            outstream.write(
+                                "{}\t{}\t{}\t{}\t{}\n".format(
+                                    filename, header, *segment
+                                )
+                            )
+            finally:
+                if filename != "-":
+                    filestream.close()
+        if args.output != "-":
+            outstream.close()
+
+    @staticmethod
+    def train(args: argparse.Namespace, options: dgmodel.Options):
+        """Train deepgrp.
+
+        Args:
+            args (argparse.Namespace): Command line arguments.
+                Includes: 'parameter', 'trainfile', 'validfile', 'bedfile',
+                'logdir', 'modelfile'
+            options (dgmodel.Options): Default and user provided options.
+        """
+        with open(args.parameter, "r") as file:
+            parameter = dgmodel.Options.from_toml(file)
+        parameter.fromdict(options.todict())
+
+        raise NotImplementedError()  # TODO:  Add training function here.
```

Comment:
Your changes need to be added at here. 
You don't need much from the Training.ipynb, just the sections 3 and 4  I think. 
Also training of one model at a time is fine and without measuring the training time, not 6 as in the notebook.